crates/consensus/src/hardforks/ecotone.rs

```diff
@@ -0,0 +1,170 @@
+//! Module containing a [Transaction] builder for the Ecotone network updgrade transactions.
+//!
+//! [Transaction]: alloy_consensus::Transaction
+
+use crate::{OpTxEnvelope, TxDeposit};
+#[cfg(not(feature = "std"))]
+use alloc::{string::String, vec, vec::Vec};
+use alloy_eips::eip2718::Encodable2718;
+use alloy_primitives::{address, bytes, Address, Bytes, TxKind, U256};
+use spin::Lazy;
+
+use crate::{hardforks::utils::upgrade_to_calldata, UpgradeDepositSource};
+
+/// The UpdgradeTo Function Signature
+pub const UPDGRADE_TO_FUNC_SIGNATURE: &str = "upgradeTo(address)";
+
+/// L1 Block Deployer Address
+pub const L1_BLOCK_DEPLOYER_ADDRESS: Address = address!("4210000000000000000000000000000000000000");
+
+/// The Gas Price Oracle Deployer Address
+pub const GAS_PRICE_ORACLE_DEPLOYER_ADDRESS: Address =
+    address!("4210000000000000000000000000000000000001");
+
+/// The new L1 Block Address
+/// This is computed by using go-ethereum's `crypto.CreateAddress` function,
+/// with the L1 Block Deployer Address and nonce 0.
+pub const NEW_L1_BLOCK_ADDRESS: Address = address!("07dbe8500fc591d1852b76fee44d5a05e13097ff");
+
+/// The Gas Price Oracle Address
+/// This is computed by using go-ethereum's `crypto.CreateAddress` function,
+/// with the Gas Price Oracle Deployer Address and nonce 0.
+pub const GAS_PRICE_ORACLE_ADDRESS: Address = address!("b528d11cc114e026f138fe568744c6d45ce6da7a");
+
+/// The Enable Ecotone Input Method 4Byte Signature
+pub const ENABLE_ECOTONE_INPUT: &[u8] = &[0x22, 0xb9, 0x08, 0xb3];
+
+/// EIP-4788 From Address
+pub const EIP4788_FROM: Address = address!("0B799C86a49DEeb90402691F1041aa3AF2d3C875");
+
+static DEPLOY_L1_BLOCK_SOURCE: Lazy<UpgradeDepositSource> =
+    Lazy::new(|| UpgradeDepositSource { intent: String::from("Ecotone: L1 Block Deployment") });
+
+static DEPLOY_GAS_PRICE_ORACLE_SOURCE: Lazy<UpgradeDepositSource> = Lazy::new(|| {
+    UpgradeDepositSource { intent: String::from("Ecotone: Gas Price Oracle Deployment") }
+});
+
+static UPDATE_L1_BLOCK_PROXY_SOURCE: Lazy<UpgradeDepositSource> =
+    Lazy::new(|| UpgradeDepositSource { intent: String::from("Ecotone: L1 Block Proxy Update") });
+
+static UPDATE_GAS_PRICE_ORACLE_SOURCE: Lazy<UpgradeDepositSource> = Lazy::new(|| {
+    UpgradeDepositSource { intent: String::from("Ecotone: Gas Price Oracle Proxy Update") }
+});
+
+static ENABLE_ECOTONE_SOURCE: Lazy<UpgradeDepositSource> = Lazy::new(|| UpgradeDepositSource {
+    intent: String::from("Ecotone: Gas Price Oracle Set Ecotone"),
+});
+
+static BEACON_ROOTS_SOURCE: Lazy<UpgradeDepositSource> = Lazy::new(|| UpgradeDepositSource {
+    intent: String::from("Ecotone: beacon block roots contract deployment"),
+});
+
+/// Builder wrapper for the Ecotone network updgrade.
+#[derive(Debug, Default, Clone, Copy)]
+pub struct EcotoneTransactionBuilder;
+
+impl EcotoneTransactionBuilder {
+    /// Constructs the Ecotone network upgrade transactions.
+    pub fn build_txs() -> Vec<Bytes> {
```

Comment:
will there be more functions on this type?

if not I'd like to keep this function here but remove the `EcotoneTransactionBuilder;` type and instead introduce a hardfork module level type that exposes this like `fn ecotone_txs() + fn fjord_txs`

Reply:
Ah that's a good idea - no there shouldn't ever be more txs on this so will take a stab at your suggestion